src/Uno.UI/UI/Xaml/Controls/Primitives/Selector.cs

```diff
@@ -683,14 +683,6 @@ private void OnItemsChanged(NotifyCollectionChangedEventArgs e)
 						ChangeSelectedItem(selectorItem, false, true);
 					}
 				}
-				else if (iVCE.CollectionChange == CollectionChange.ItemRemoved)
-				{
-					// If the removed item is the currently selected one, Set SelectedIndex to -1
-					if ((int)iVCE.Index == SelectedIndex)
-					{
-						ResetIndexIfNeeded();
-					}
-				}
```

Comment:
`OnItemsSourceSingleCollectionChanged` (which is the only call site for this function) already calls `ResetIndexIfNeeded`, so calling it again here causes the bug.

Reply:
Let's make sure that the tests from https://github.com/unoplatform/uno/pull/9340, https://github.com/unoplatform/uno/pull/6742 and https://github.com/unoplatform/uno/pull/4096 are still passing.

Reply:
CI is green and the manual tests from #4096 look fine.